### ISSUE TITLE
Add option to ignore arrow function arguments in `id-length` rule

### ISSUE THREAD

**Issue by joeyespo** [eslint/eslint#9006]
**Tell us about your environment**

* **ESLint Version: 4.3.0**
* **Node Version: 8.1.1**
* **npm Version: 5.3.0**

**What parser (default, Babel-ESLint, etc.) are you using?**

Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "root": true,
  "extends": "eslint:recommended",
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module"
  },
  "env": {
    "es6": true,
    "node": true
  },
  "rules": {
    "no-extra-parens": "error",
    "arrow-parens": ["error", "as-needed", { "requireForBlockBody": true }],
    "no-unused-vars": ["error", { "args": "none" }],
    "id-length": ["error", { "properties": "never" }]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
    const result = items
      .where(x => x.amount > 0)
      .sort((a, b) => strcmp(a.name, b.name));
```

**What did you expect to happen?**

I expect a way to allow arrow function params to be excluded from the "min" ID length. Similar to #3450, 
these can be exceptional cases. Unlike variables elsewhere, the context is often clear and using more characters can quickly become verbose in LINQ-like usage. In the meantime, I'm considering turning this rule off, but that's still not ideal because then other variables can unintentionally be left too short.

**What actually happened? Please include the actual, raw output from ESLint.**

```
path-to-my-file.js
  63:14  error  Identifier name 'x' is too short (< 2)    id-length
  64:14  error  Identifier name 'a' is too short (< 2)    id-length
  64:17  error  Identifier name 'b' is too short (< 2)    id-length
```

My suggestion is to add an option called `arrow-params` and, like `properties`, accept `"never"` to turn the rule off for this case. This beats having to turn off the rule or

Let me know if you need any more context or info. Thank you!

**Comment by platinumazure** [comment]
Hi @joeyespo, thanks for the issue! I think this could be a reasonable enhancement.

Thanks very much for taking the time to fill out our bug report template. Would you mind using this [rule enhancement proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md) instead, with your thoughts of the best way to change the rule schema to account for your use case?

My only concern is whether we should also consider supporting all function expression arguments (for those who are not using arrow functions as callbacks), vs just arrow function expressions. But maybe a resolution on that will shake out after some discussion.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
